### PR TITLE
[chrome-ext] lazy load charts to improve dashboard render time

### PR DESCRIPTION
**Pull request by runpengl**
Chart render no longer blocks rendering of the other dashboard components (which now load almost immediately)

### CODE REVIEW

File: packages/chrome-extension/src/tsconfig.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
       "ES2017.Object",
     ],
     "moduleResolution": "node",
+    "module": "esnext",
```
Comment (runpengl) [inline packages/chrome-extension/src/tsconfig.json]: this was required to get compiler to stop complaining about usage of `React.lazy`

File: packages/chrome-extension/src/dashboard/charts/SolveProgressChart.js [hunk 2]
```diff
@@ -176,7 +178,7 @@ export default class SolveProgressChart extends Component {
             grid: {
                 top: '5%',
                 left: '3%',
-                right: '15%',
+                right: '110px',
```
Comment (runpengl) [inline packages/chrome-extension/src/dashboard/charts/SolveProgressChart.js]: ensures legend never overlaps with grid on smaller-width screens

File: packages/chrome-extension/webpack.config.js [hunk 3]
```diff
@@ -1,6 +1,5 @@
 const MiniCssExtractPlugin = require("mini-css-extract-plugin");
 const path = require("path");
-const webpack = require("webpack");
```
Comment (runpengl) [inline packages/chrome-extension/webpack.config.js]: this was unused

File: packages/chrome-extension/webpack.config.js [hunk 4]
```diff
@@ -30,6 +29,7 @@ const config = {
     mode: production ? "production" : "development",
     output: {
         path: path.resolve(__dirname, "dist"),
+        publicPath: "/",
```
Comment (runpengl) [inline packages/chrome-extension/webpack.config.js]: ensures lazy-loaded chunks have the correct path

**Comment by runpengl** [comment]
> @jeevnayak brought up a good point that we'd want to maximize real estate to view the puzzles (when using the dashboard on our computers). he's exploring defaulting the chart to a collapsed state, which means that we won't need to render a chart immediately on page load. would that fix the perf problem?

I like the idea of making the chart dismissible. I believe it would address the perf problem without lazy-loading only if the chart defaults in a collapsed state. But in the current state of @jeevnayak's `chart` branch (where chart is shown on page load), the chart render is still blocking.

**Comment by runpengl** [comment]
> what was actually the bottleneck? Was it just rendering `SolveProgressChart` (and not loading/parsing the charting library)? We didn't actually need to use `React.lazy` and bundle splitting to solve this problem, right?

Yeah, it's the chart rendering which requires loading and parsing of the library to be completed first. the lazy-load allows other components on the page to render without waiting for the chart and its dependencies.

**Comment by vsiao** [comment]
> But in the current state of @jeevnayak's chart branch (where chart is shown on page load), the chart render is still blocking.

yeah, that's what it is right now. we did discuss defaulting it to collapsed though—I think he's following up with that

**Comment by jeevnayak** [comment]
> > But in the current state of @jeevnayak's chart branch (where chart is shown on page load), the chart render is still blocking.
> 
> yeah, that's what it is right now. we did discuss defaulting it to collapsed though—I think he's following up with that

Yep, just wanted to check with you first @runpengl to make sure that was okay - I'll make the change to default to collapsed

**Pull request closed without merging**